framework/audio/midi/internal/sequencer.cpp

```diff
@@ -58,6 +49,53 @@ void Sequencer::init(float samplerate, float gain)
     });
 }
 
+void Sequencer::loadMIDI(const std::shared_ptr<MidiStream>& stream)
+{
+    m_midiStream = stream;
+    m_streamState = StreamState();
+
+    m_midiData = stream->initData;

```

Comment:
I' curious, why is it **init** data?

Reply:
The idea is that `MidiStream` should have a data for initialization, which should at least contain information about tracks and channels (midi channels), there may or may not be midi events.